components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/TenantWorkflowConfigHolder.java

```diff
@@ -159,7 +159,14 @@ public void load() throws WorkflowException, RegistryException {
 
             workflowElem = workflowExtensionsElem.getFirstChildWithName(
                     new QName(WorkflowConstants.APPLICATION_DELETION));
-            executorClass = workflowElem.getAttributeValue(new QName(WorkflowConstants.EXECUTOR));
+            if (workflowElem == null) {
+                workflowElem = OMAbstractFactory.getOMFactory()
+                        .createOMElement(new QName(WorkflowConstants.API_STATE_CHANGE));
+                executorClass = WorkflowConstants.DEFAULT_EXECUTOR_API_STATE_CHANGE;
```

Comment:
Why are we setting default API state change executor as the default one when APPLICATION_DELETION workflow is missing?